shiny/ui/_chat.py

```diff
@@ -50,6 +56,37 @@
     "ChatMessageDict",
 )
 
+if TYPE_CHECKING:
+
+    from chatlas import Chat as ChatlasClient
+    from chatlas import Turn as ChatlasTurn
+
+    # ChatlasClient
+    has_chatlas = True
+else:
+    ChatlasTurn = ChatlasClient = object
+    has_chatlas = False
+
+
+@runtime_checkable
+class ClientWithState(Protocol):
+    async def get_state(self) -> Jsonifiable: ...
+    async def set_state(self, value: Jsonifiable): ...
```

Comment:
Add a docstring to explaining a bit what we're expecting `get_state`/`set_state` to do.